src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs

```diff
@@ -24,13 +26,56 @@ public AsyncSessionDocumentTimeSeries(InMemoryDocumentSessionOperations session,
         }
         public async Task<IEnumerable<TimeSeriesValue>> GetAsync(string timeseries, DateTime from, DateTime to, CancellationToken token = default)
         {
-            if (Session.TimeSeriesByDocId.TryGetValue(DocId, out var cache))
+            TimeSeriesDetails details;
+            
+            if (Session.TimeSeriesByDocId.TryGetValue(DocId, out var cache) &&
+                cache.TryGetValue(timeseries, out var ranges))
```

Comment:
Add a check here that the `ranges` is not empty

Reply:
done